apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -256,6 +289,39 @@ function wpcom_global_styles_in_use() {
 	return $global_styles_in_use;
 }
 
+/**
+ * Checks if Global Styles has been customized by the given site in the past.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether Global Styles have been customized.
+ */
+function wpcom_global_styles_has_been_used( $blog_id = 0 ) {
```

Comment:
I have mixed feelings about using this here. This might cause some performance issues since this code should run on the front-end too, which means regular site visitors will hit this code path. 

Now, even though this should run for each site only once, it might cause a spike of queries at once, after the code lands. 

Also, since the sticker is added for just one type of site (sites that should be limited), it means that this query will run on every request for the other type of sites, since we are not caching the result for those sites. So, to sum it up:
 - For sites limited this will run only once
 - For sites that are not limited, this code will run on every request since we are not caching the result with a blog sticker.

Reply:
Since `get_posts` is a read operation, it's probably cached - although `wpcom_global_styles_in_use_by_wp_global_styles_post` looks a bit heavy (since we are doing a json decode and some array manipulations). My concern is related to `add_blog_sticker` since that should be a write operation (most likely uncached).

Reply:
I'm not super concerned about the spike, as TBH I would expect most sites to only have a handful of GS posts.

But you make it a good point: the sticker doesn't seem a good way to store this info, as we need to know both the positive and negative states, not just its existence. 🤔 

I was thinking about site options, but would rather find something more private if possible 🤔